src/cluster_legacy.c

```diff
@@ -2607,7 +2607,7 @@ int nodeUpdateAddressIfNeeded(clusterNode *node, clusterLink *link, clusterMsg *
     /* Check if this is our primary and we have to change the
      * replication target as well. */
     if (nodeIsReplica(myself) && myself->replicaof == node)
-        replicationSetPrimary(node->ip, getNodeDefaultReplicationPort(node), 0);
+        replicationSetPrimary(node->ip, getNodeDefaultReplicationPort(node), 0, 1);
```

Comment:
same as before. in this flow I am not sure we should disco blocked clients. we were replica before and we will not perform re-sync.